_posts/2020-05-30-Metal3-Cluster-Deployment.md

```diff
@@ -0,0 +1,544 @@
+      Discovery:
+      Node Registration:
+        Kubelet Extra Args:
+          Node - Labels:  metal3.io/uuid={{ ds.meta_data.uuid }}
+        Name:             {{ ds.meta_data.name }}
+    Post Kubeadm Commands:
+      mkdir -p /home/metal3/.kube
+      cp /etc/kubernetes/admin.conf /home/metal3/.kube/config
+      chown metal3:metal3 /home/metal3/.kube/config
+    Pre Kubeadm Commands:
+      ifup eth1
+      dnf update -y
+      dnf install -y ebtables socat conntrack-tools
+      dnf install python3 -y
+      dnf config-manager --add-repo https://download.docker.com/linux/centos/docker-ce.repo
+      setenforce 0
+      sed -i 's/^SELINUX=enforcing$/SELINUX=permissive/' /etc/selinux/config
+      dnf install docker-ce docker-ce-cli --disableexcludes=kubernetes --nobest -y
+      dnf install gcc kernel-headers kernel-devel keepalived device-mapper-persistent-data lvm2 -y
+      echo  "Installing kubernetes binaries"
+      curl -L --remote-name-all https://storage.googleapis.com/kubernetes-release/release/v1.18.0/bin/linux/amd64/{kubeadm,kubelet,kubectl}
+      chmod a+x kubeadm kubelet kubectl
+      mv kubeadm kubelet kubectl /usr/local/bin/
+      mkdir -p /etc/systemd/system/kubelet.service.d
+      curl -sSL "https://raw.githubusercontent.com/kubernetes/release/v0.2.7/cmd/kubepkg/templates/latest/deb/kubelet/lib/systemd/system/kubelet.service" | sed "s:/usr/bin:/usr/local/bin:g" > /etc/systemd/system/kubelet.service
+      curl -sSL "https://raw.githubusercontent.com/kubernetes/release/v0.2.7/cmd/kubepkg/templates/latest/deb/kubeadm/10-kubeadm.conf" | sed "s:/usr/bin:/usr/local/bin:g" > /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
+      usermod -aG docker metal3
+      systemctl enable --now docker keepalived kubelet
+    Users:
+      Name:  metal3
+      Ssh Authorized Keys: ${SSH_KEY}
+      Sudo:  ALL=(ALL) NOPASSWD:ALL
+  Replicas:  1
+  Version:   v1.18.0
+Status:
+  Replicas:              1
+  Selector:              cluster.x-k8s.io/cluster-name=eko,cluster.x-k8s.io/control-plane=
+  Unavailable Replicas:  1
+  Updated Replicas:      1
+Events:                  <none>
+```
+
+```console
+kubectl get Metal3MachineTemplate ${CLUSTER_NAME}-controlplane -n metal3 
+kubectl get bmh -n metal3 -w   > You should see all the nodes that were created at the time of metal3 deployment, along with their current status as the provisioning progresses
+```
+
+> Once the provisioning is finished, let's get the host-ip : 
+```console
+sudo virsh net-dhcp-leases baremetal
+```
+> **Note** : *baremetal is one of the 2 networks that were created at the time of Metal3 deployment, the other being “provisioning” which is used - as you have guessed - for provisioning the bare metal cluster. More details about networking setup in the metal3-dev-env environment are described in a previous [blog post](http://baremetal).*
+
+You can login to the master node if you want, and can check the deployment status
+```console
+ssh metal3@{master-node-ip}
```

Comment:
that covers the login part.

How do I check the deployment status?

Mention how the configuration gets applied and how it becomes a Kubernetes master